File: app/android/app/src/main/java/bosch/smartcampus/thermalcomfortstudy/packet/GsrItem.java
package bosch.smartcampus.thermalcomfortstudy.packet;

import bosch.smartcampus.thermalcomfortstudy.lib.Timestamp;

/**
 * Created by rsukkerd on 6/27/16.
 */
public class GsrItem extends BandDataItem {
    private static final String GSR_NAME = "Gsr";

    private double gsr;

    public GsrItem(String username, Timestamp timestamp,
                   double gsr) {
        super(username, timestamp);
        this.gsr = gsr;
    }

    public double getGsr() {
        return gsr;
    }

    @Override
    public String toXML() {
        StringBuilder builder = new StringBuilder();
        builder.append("<" + ELEMENT + " id=\"" + getId() + "\">");
        builder.append(getTransducerDataElement(GSR_NAME, Double.toString(gsr)));
        builder.append("</" + ELEMENT + ">");
        return builder.toString();
    }
}

File: app/iOS/Thermal Comfort Study/GSRItem.swift
//
//  GSRItem.swift
//  Thermal Comfort Study
//
//  Created by Madhav Achar on 10/14/16.
//  Copyright © 2016 boschsc. All rights reserved.
//

import Foundation

class GSRItem: BandDataItem {
    
    private static let GSR_NAME = "Gsr"
    private var gsr: Double!
    
    init(username: String, timestamp: Date, gsr: Double) {
        super.init(username, timestamp)
        self.gsr = gsr
    }
    
    func toXML() -> String {
        return super.toXML(dataName: type(of: self).GSR_NAME,
                           value: self.gsr)
    }
}
